src/Tools/dotnet-monitor/OpenApi/OpenApiOptionsExtensions.cs

```diff
@@ -0,0 +1,239 @@
+                if (type == typeof(CaptureParametersConfiguration) || 
+                    type == typeof(CollectionRuleDetailedDescription) ||
+                    type == typeof(DotnetMonitorInfo) ||
+                    type == typeof(EventMetricsConfiguration) ||
+                    type == typeof(EventMetricsMeter) ||
+                    type == typeof(EventMetricsProvider) ||
+                    type == typeof(EventPipeConfiguration) ||
+                    type == typeof(EventPipeProvider) ||
+                    type == typeof(ExceptionFilter) ||
+                    type == typeof(ExceptionsConfiguration) || 
+                    type == typeof(MethodDescription) ||
+                    type == typeof(MonitorCapability) ||
+                    type == typeof(OperationError) ||
+                    type == typeof(OperationProcessInfo) ||
+                    type == typeof(OperationStatus) ||
+                    type == typeof(OperationSummary) ||
+                    type == typeof(ProcessIdentifier) ||
+                    type == typeof(ProcessInfo))
+                {
+                    schema.AdditionalPropertiesAllowed = false;
+                }
+                else if (type == typeof(ProblemDetails) ||
+                         type == typeof(ValidationProblemDetails))
+                {
+                    schema.AdditionalProperties = new OpenApiSchema();
+                }
+                else if (type == typeof(LogsConfiguration))
+                {
+                    schema.AdditionalPropertiesAllowed = false;
+                    schema.Properties["filterSpecs"].AdditionalProperties = new OpenApiSchema
+                    {
+                        Reference = new OpenApiReference
+                        {
+                            Type = ReferenceType.Schema,
+                            Id = nameof(LogLevel)
+                        }
+                    };
+                }
+                return Task.CompletedTask;
+            });
+
+            // Fix up "additionalProperties" for the response type of GetCollectionRulesDescription
+            options.AddOperationTransformer((operation, context, cancellationToken) => {
+                if (operation.OperationId == nameof(DiagController.GetCollectionRulesDescription))
+                {
+                    foreach (var response in operation.Responses)
+                    {
+                        if (response.Key == StatusCodeStrings.Status200Ok)
+                        {
+                            var schema = response.Value.Content[ContentTypes.ApplicationJson].Schema;
+                            schema.AdditionalProperties.AdditionalPropertiesAllowed = false;
+                        }
+                    }
+                }
+                return Task.CompletedTask;
+            });
+
+            // Add missing descriptions on types
+            options.AddSchemaTransformer((schema, context, cancellationToken) => {
```

Comment:
Could these descriptions be added directly to the types with the DescriptionAttribute? If those are not detected, then could you change this transformer to check if associated schema type has a DescriptionAttribute and apply its value to the schema? This would make it a more generic solution.